aspnetcore/mvc/models/file-uploads/samples/2.x/SampleApp/Utilities/FileHelpers.cs

```diff
@@ -0,0 +1,268 @@
+                    {
+                        modelState.AddModelError(formFile.Name,
+                            $"{fieldDisplayName}({trustedFileName}) file " +
+                            "type isn't permitted or the file's signature " +
+                            "doesn't match the file's extension.");
+                    }
+                    else
+                    {
+                        return memoryStream.ToArray();
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                modelState.AddModelError(formFile.Name,
+                    $"{fieldDisplayName}({trustedFileName}) upload failed. " +
+                    $"Please contact the Help Desk for support. Error: {ex.HResult}");
+                // Log the exception
+            }
+
+            return new byte[0];
+        }
+
+        public static async Task<byte[]> ProcessStreamedFile(
+            MultipartSection section, ContentDispositionHeaderValue contentDisposition,
+            ModelStateDictionary modelState, string[] permittedExtensions, long sizeLimit)
+        {
+            try
+            {
+                using (var memoryStream = new MemoryStream())
+                {
+                    await section.Body.CopyToAsync(memoryStream);
+
+                    // Check if the file is empty or exceeds the size limit.
+                    if (memoryStream.Length == 0)
+                    {
+                        modelState.AddModelError("File", "The file is empty.");
+                    }
+                    else if (memoryStream.Length > sizeLimit)
+                    {
+                        var megabyteSizeLimit = sizeLimit / 1048576;
+                        modelState.AddModelError("File",
+                        $"The file exceeds {megabyteSizeLimit:N1} MB.");
+                    }
+                    else if (!IsValidFileExtensionAndSignature(
+                        contentDisposition.FileName.Value, memoryStream,
+                        permittedExtensions))
+                    {
+                        modelState.AddModelError("File",
+                            "The file type isn't permitted or the file's " +
+                            "signature doesn't match the file's extension.");
+                    }
+                    else
+                    {
+                        return memoryStream.ToArray();
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                modelState.AddModelError("File",
+                    "The upload failed. Please contact the Help Desk " +
+                    $" for support. Error: {ex.HResult}");
+                // Log the exception
+            }
+
+            return new byte[0];
+        }
+
+        private static bool IsValidFileExtensionAndSignature(string fileName, Stream data, string[] permittedExtensions)
```

Comment:
What are you trying to do here? This has become more than a sample and started to veer into security sensitive guidance, which feels to me to be very out of scope for docs.